validation/util/test.go

```diff
@@ -184,3 +248,74 @@ func RuntimeOutsideValidate(g *generate.Generator, f AfterFunc) error {
 	}
 	return nil
 }
+
+// RuntimeLifecycleValidate validates runtime lifecycle.
+func RuntimeLifecycleValidate(g *generate.Generator, lifecycle *LifecycleFuncs, actions LifecycleAction) error {
```

Comment:
I don't see a need to split funcs from actions.  Can we rename `LifecycleFuncs` to `LifecycleConfig` (or just `Lifecycle`?) and add an `Actions` property?

Reply:
+1 for this